src/types/__snapshots__/resolveDescriptors.spec.ts.snap

```diff
@@ -315,6 +315,14 @@ exports[`resolveDescriptors should fail descriptors for contract-receiver-struct
 "
 `;
 
+exports[`resolveDescriptors should fail descriptors for contract-with-self-inheritance 1`] = `
+"<unknown>:1:1: Self-inheritance detected in the list of inherited traits for contract "Example"
```

Comment:
I'd change the error message to something more simple, like `Self-inheritance is not allowed`